code/__HELPERS/lists.dm

```diff
@@ -344,3 +344,13 @@
 	while(L.Remove(null))
 		continue
 	return L
+
+//Copies a list, and all lists inside it recusively
+//Does not copy any other reference type
+/proc/deepCopyList(list/l)
+	if(!islist(l))
+		return l
+	. = l.Copy()
+	for(var/i = 1 to length(.))

```

Comment:
You're right, `.len` is a hair faster, which surprised me since `length()`, like most built-ins, is its own op code.  Nevertheless, this is only called once per list, so I don't think it warrants another commit.